src/main/java/hanghaeboard/api/service/board/BoardService.java

```diff
@@ -0,0 +1,87 @@
+package hanghaeboard.api.service.board;
+
+import hanghaeboard.api.controller.board.request.CreateBoardRequest;
+import hanghaeboard.api.controller.board.request.DeleteBoardRequest;
+import hanghaeboard.api.controller.board.request.UpdateBoardRequest;
+import hanghaeboard.api.exception.exception.InvalidPasswordException;
+import hanghaeboard.api.service.board.response.CreateBoardResponse;
+import hanghaeboard.api.service.board.response.DeleteBoardResponse;
+import hanghaeboard.api.service.board.response.FindBoardResponse;
+import hanghaeboard.api.service.board.response.UpdateBoardResponse;
+import hanghaeboard.domain.board.Board;
+import hanghaeboard.domain.board.BoardRepository;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class BoardService {
+
+    private final BoardRepository boardRepository;
+
+    @Transactional
+    public CreateBoardResponse createBoard(CreateBoardRequest request) {
+
+        Board board = request.toEntity();
+        Board savedBoard = boardRepository.save(board);
+
+        return CreateBoardResponse.from(savedBoard);
+    }
+
+    public List<FindBoardResponse> findAllBoard(){
+        return boardRepository.findAllBoard();
+    }
+
+    public FindBoardResponse findBoardById(Long id){
+
+        Board findBoard = boardRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("조회된 게시물이 없습니다."));
+
+        if(findBoard.isDeleted()){
+            throw new EntityNotFoundException("삭제된 게시물입니다.");
+        }
+
+        return FindBoardResponse.from(findBoard);
+    }
+
+    @Transactional
+    public UpdateBoardResponse updateBoard(Long id, UpdateBoardRequest request) {
+        Board findBoard = findBoard(id, request.getPassword());
+
+        if(findBoard.isDeleted()){
+            throw new EntityNotFoundException("삭제된 게시물입니다.");
+        }
+
+        findBoard.changeBoard(request.getWriter(), request.getTitle(), request.getContent());
+
+        return UpdateBoardResponse.from(findBoard);
+    }
+
+    private Board findBoard(Long id, String request) {
+        Board findBoard = boardRepository.findById(id).orElseThrow(() -> new EntityNotFoundException("조회된 게시물이 없습니다."));
+
+        if (!findBoard.isCorrectPassword(request)) {
+            throw new InvalidPasswordException("비밀번호가 올바르지 않습니다.");
+        }
+        return findBoard;
```

Comment:
부정연산자보다 메서드명을 리팩토링 하는 것이 가독성에 더 좋아보이는데 건엽님 생각은 어떠신가요!? 🙃

```suggestion
        if (findBoard.isNotCorrectPassword(request)) {
            throw new InvalidPasswordException("비밀번호가 올바르지 않습니다.");
        }
        return findBoard;
```

Reply:
좋은 의견 같습니다 !